src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -205,6 +205,12 @@ p5.RendererGL.prototype.endShape = function(
   this._processVertices(...arguments);
```

Comment:
Actually, I just noticed that this is where the triangle tesselation happens, so we would need to move your if statement up so that it happens before this line (right above line 204 is I think where it should go.)